TACC/Core-Portal
# FP-456: Load Latest CMS Header Styles & Pin CMS and Docs Images

## Overview: ##

Load the correct __new__ header stylesheet from the CMS.

Also, pin the images of Docker services `cms` and `docs` to use tags that match updated versions of each. 

## PR Status: ##


## Related Jira tickets: ##

* [FP-456](https://jira.tacc.utexas.edu/browse/FP-456)

## Summary of Changes: ##

- **Key Change:** Fix the out-of-date path to the header stylesheet.
- **Suggested Change** Add a `README.md` to `/server` that, most relevantly, explains the difference between the two pairs of `static` and `media` directories.
- **DevOps Change** Update the build pointers for {{cms}} and {{docs}} services to newest compatible builds.

## Testing Steps: ##

### 1. Setup

Choose either method.

#### A. Fresh Local Install of Portal

1. Checkout the branch of this PR.
1. Set up a fresh install of Portal _(by following the instructions in the repo `README.md`)_.
1. Ensure all dockers are running.

#### B. Existing Local Install of Portal

1. Checkout the branch of this PR.
1. In a terminal, run `docker-compose -f ./server/conf/docker/docker-compose-dev.all.debug.yml up --build --force-recreate --attach-dependencies cms`
1. In another Terminal, run:
    1. `docker exec -it frontera_prtl_cms /bin/bash`
    1. `python3 manage.py migrate`
        - _(added during review, but testing was done with Setup method A)_
    1. `python3 manage.py collectstatic`
1. In a terminal, run `docker-compose -f ./server/conf/docker/docker-compose-dev.all.debug.yml up --build --force-recreate --attach-dependencies docs`
1. Ensure all dockers are running.

### 2. Test

1. Load Portal dashboard, and ensure:
    - header renders like production
    - the loaded `site.header.css` is from `/static/build/styles/site.header.css`
1. Load CMS website, and ensure:
    - header renders like production
    - the loaded `site.header.css` is from `/static/build/styles/site.header.css`
1. Load User Guide minisite, and ensure:
    - header renders like production
    - the loaded `site.header.css` is from `/static/build/styles/site.header.css`

## UI Photos:

![FP-456 User Guide](https://user-images.githubusercontent.com/62723358/84536062-ac1d9f00-acb2-11ea-82c5-9980034907a3.png)
![FP-456 CMS Website](https://user-images.githubusercontent.com/62723358/84536063-acb63580-acb2-11ea-9fc2-e5bd8010b921.png)
![FP-456 Portal Client](https://user-images.githubusercontent.com/62723358/84536064-ad4ecc00-acb2-11ea-91a6-6cc992736d5c.png)

## Notes: ##

This is similar to PR #64, but relies on up-to-date (not production) CMS and Docs code.
